ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/ImageSingleVulnerabilities.tsx

```diff
@@ -1,13 +1,208 @@
-import React from 'react';
-import { PageSection, Text, Divider } from '@patternfly/react-core';
+import React, { ReactNode } from 'react';
+import {
+    Bullseye,
+    Divider,
+    EmptyState,
+    EmptyStateBody,
+    EmptyStateIcon,
+    EmptyStateVariant,
+    Grid,
+    GridItem,
+    PageSection,
+    Spinner,
+    Tab,
+    TabTitleText,
+    Tabs,
+    TabsComponent,
+    TabsProps,
+    Text,
+    Title,
+} from '@patternfly/react-core';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+import { gql, useQuery } from '@apollo/client';
+
+import { VulnerabilitySeverity } from 'types/cve.proto';
+import { getAxiosErrorMessage } from 'utils/responseErrorUtils';
+import { FixableStatus, isValidCveStatusTab } from './types';
+import useCveStatusTabParameter from './hooks/useCveStatusTabParameter';
+import WorkloadTableToolbar from './WorkloadTableToolbar';
+import BySeveritySummaryCard from './SummaryCards/BySeveritySummaryCard';
+import CvesByStatusSummaryCard from './SummaryCards/CvesByStatusSummaryCard';
+
+function severityCountsFromImageVulnerabilities(
+    imageVulnerabilities: ImageVulnerabilitiesResponse['image']['imageVulnerabilities']
+): Record<VulnerabilitySeverity, number> {
+    const severityCounts = {
+        LOW_VULNERABILITY_SEVERITY: 0,
+        MODERATE_VULNERABILITY_SEVERITY: 0,
+        IMPORTANT_VULNERABILITY_SEVERITY: 0,
+        CRITICAL_VULNERABILITY_SEVERITY: 0,
+    };
+
+    imageVulnerabilities.forEach(({ severity }) => {
+        severityCounts[severity] += 1;
+    });
+
+    return severityCounts;
+}
+
+function statusCountsFromImageVulnerabilities(
+    imageVulnerabilities: ImageVulnerabilitiesResponse['image']['imageVulnerabilities']
+): Record<FixableStatus, number> {
+    const statusCounts = {
+        Fixable: 0,
+        'Not fixable': 0,
+    };
+
+    imageVulnerabilities.forEach(({ isFixable }) => {
+        if (isFixable) {
+            statusCounts.Fixable += 1;
+        } else {
+            statusCounts['Not fixable'] += 1;
+        }
+    });
+
+    return statusCounts;
+}
```

Comment:
can we pull these under the type defs so it's easier to scan? i personally like having the types be defined first, but if this makes sense to you i'm open to that too

Reply:
Sure that works for me. How about types for the React component props? I've been trying to keep the prop type definitions directly above the component, but below other functions and type definitions.